Kickstarter-iOS/Views/Controllers/RewardsCollectionViewController.swift

```diff
@@ -94,23 +88,17 @@ final class RewardsCollectionViewController: UICollectionViewController {
 
     guard let layout = self.flowLayout else { return }
 
-    self.updateHiddenScrollViewBoundsIfNeeded(for: layout)
-  }
+    let itemSize = self.calculateItemSize(from: layout, using: self.collectionView)
 
-  override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
-    super.viewWillTransition(to: size, with: coordinator)
-
-    self.flowLayout?.invalidateLayout()
+    if itemSize != layout.itemSize {
```

Comment:
Is it necessary to compare the sizes? Trying to understand what's going on in here...

Reply:
It is. Basically during one of the layout passes, the `adjustedContentInsets` change, and our `itemSize` depends on those `contentInset` values. In order to make sure the `itemSize` has all the update-to-date values (and so that it's the right height), we need to do this check.